kie-api/src/main/java/org/kie/api/internal/utils/ServiceDiscoveryImpl.java

```diff
@@ -126,7 +113,12 @@ private void processKieService(ClassLoader classLoader, String key, String value
                     childServices.computeIfAbsent( serviceName, k -> new ArrayList<>() )
                             .add( newInstance( classLoader, value.substring( 1 ) ) );
                 } else {
-                    services.put( serviceName, newInstance( classLoader, value ) );
+                    int priority = 0;
+                    if ( value.charAt( 0 ) >= '0' && value.charAt( 0 ) <= '9' ) {
```

Comment:
extract value.charAt(0)

Also what does this function do? it's unreadable IMHO